server/pulp/server/content/web/views.py

```diff
@@ -137,10 +137,13 @@ def get(self, request):
         host = request.get_host()
         path = os.path.realpath(request.path_info)
 
-        # Authorization
-        if not allow_access(request.environ, host):
-            # Not Authorized
-            return HttpResponseForbidden()
+        # Check authorization if http isn't being used. This environ variable must
+        # be available in all implementations so it is not dependant on Apache httpd:
+        # https://www.python.org/dev/peps/pep-0333/#environ-variables
+        if request.environ['wsgi.url_scheme'] != 'http':
```

Comment:
This isn't great, but it's this Apache snippet in the plugin (just `pulp_rpm` right now):
```
<Location /pulp/repos/>
  RewriteEngine On
  RewriteCond %{HTTPS} on
  RewriteRule (.+/pulp/repos/)(.*) /pulp/content/var/www/pub/yum/https/repos/$2 [DPI]
  RewriteCond %{HTTPS} off
  RewriteRule (.+/pulp/repos/)(.*) /pulp/content/var/www/pub/yum/http/repos/$2 [DPI]
</Location>
```
When we check the existence of `/var/www/pub/yum/http/repos/...` (which is the value of `path_info`) it won't exist unless you've published a protected repo to HTTP in which case wat?

I don't like that the security of this is held together by shoestrings and gum, but so it goes.